### PR TITLE
feat(auth-server): add chargeCustomer to Paypalhelper

### PR DESCRIPTION
**Pull request by birdsarah**
## Because

- We need to make a call to PayPal's DoReferenceTransaction to charge a customer

## This pull request

- Adds a doReferenceTransaction method to Paypal client
- Adds a chargeCustomer to PaypalHelper
- Adds fixtures for DoReferenceTransaction and CreateBillingAgreement
- Small clean-up, renames SetCheckout to SetExpressCheckout for consistency

## Issue that this pull request solves

Closes: #7236

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)

Please attach the screenshots of the changes made in case of change in user interface.

## Other information (Optional)

Any other information that is important to this pull request.

### CODE REVIEW

File: packages/fxa-auth-server/lib/payments/paypal-client.ts [hunk 1]
```diff
@@ -4,6 +4,7 @@
 import pRetry from 'p-retry';
 import querystring from 'querystring';
 import superagent from 'superagent';
+import { ChargeCustomerOptions } from './paypal';
```
Comment (bbangert) [inline packages/fxa-auth-server/lib/payments/paypal-client.ts:7]: It doesn't seem like the right separation of concerns to have a lower level client pull in types from the higher level helper library for its own use. Ideally the lower level client could be yanked out of FxA entirely on its own for use as a stand-alone library (though obviously we aren't going to do that at this time).

**Review by bbangert — CHANGES_REQUESTED** [review]
Looks great to me, only change request is re: the type location.

File: packages/fxa-auth-server/lib/payments/paypal-client.ts (hunk 1 above)
Comment (birdsarah) [reply packages/fxa-auth-server/lib/payments/paypal-client.ts:7]: I agree. Will change.

**Review by bbangert — APPROVED** [review]
Nice! 💯

**Pull request merged**